homeassistant/components/rest/switch.py

```diff
@@ -55,7 +59,8 @@
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         **TEMPLATE_ENTITY_BASE_SCHEMA.schema,
-        vol.Required(CONF_RESOURCE): cv.url,
+        vol.Exclusive(CONF_RESOURCE, CONF_RESOURCE): cv.url,
+        vol.Exclusive(CONF_RESOURCE_TEMPLATE, CONF_RESOURCE): cv.template,
```

Comment:
IMO this should be refactored in a similar way to the `binary_sensor` and `sensor` platform, by extending from `RESOURCE_SCHEMA`.

`PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({**RESOURCE_SCHEMA, **SWITCH_SCHEMA})`

Reply:
Done, but not exactly like that because it was not working : RESOURCE_SCHEMA has a default conf_method of "GET", while rest switch has a default of "post".

I ended up doing so : `PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({**RESOURCE_SCHEMA}).extend({**SWITCH_SCHEMA})`

NOTE : I noticed that switch methods are all lowercase, while the rest of sensors require uppercase. Changing that would be a breaking change, so I'll leave it here for now, that's not the point of this P.R.

NOTE 2 : I've not yet removed code for Platformnotready, I'm waiting the other P.R. https://github.com/home-assistant/core/pull/90627  to be validated to see how to do it (if any change is required), because I need to update resource prior to raising exception, so as soon as P.R. is merged, the change I've done are "required".

Reply:
done